include/device.h

```diff
@@ -497,6 +497,47 @@ device_required_handles_get(const struct device *dev,
 	return rv;
 }
 
+/**
+ * @brief Get the set of handles that this device supports.
+ *
+ * The set of supported devices is inferred from devicetree, and does not
+ * include any software constructs that may depend on the device.
+ *
+ * @param dev the device for which supports are desired.
+ *
+ * @param count pointer to a place to store the number of devices provided at
+ * the returned pointer. The value is not set if the call returns a null
+ * pointer. The value may be set to zero.
+ *
+ * @return a pointer to a sequence of @p *count device handles, or a null
+ * pointer if @p dh does not provide dependency information.
+ */
+static inline const device_handle_t *
+device_supported_handles_get(const struct device *dev,
```

Comment:
Can we rethink the name? "supported" is really not very explanatory here. What exactly are we returning here, from the point of view of the Devicetree?

Reply:
Could we use "children" here instead perhaps?

Reply:
The existing API usage is `SUPPORTS_ORDS`, and as this is the exact same information just in a different forms, I don't believe the naming should be different. Whether the existing API should be changed I have no strong feelings on.
https://github.com/zephyrproject-rtos/zephyr/blob/0d4f685dbdb521a35abbb2ad764b775eb3ecdc66/include/devicetree/ordinals.h#L46-L61

Reply:
OK, no objections in that case @JordanYates